Documentation/cloud-config.md

```diff
@@ -39,32 +39,34 @@ CoreOS tries to conform to each platform's native method to provide user data. E
 
 ### coreos
 
-#### etcd
+#### etcd2
 
-The `coreos.etcd.*` parameters will be translated to a partial systemd unit acting as an etcd configuration file.
+The `coreos.etcd2.*` parameters will be translated to a partial systemd unit acting as an etcd configuration file.
 If the platform environment supports the templating feature of coreos-cloudinit it is possible to automate etcd configuration with the `$private_ipv4` and `$public_ipv4` fields. For example, the following cloud-config document...
 
 ```yaml
 #cloud-config
 
 coreos:
-  etcd:
-      name: node001
-      # generate a new token for each unique cluster from https://discovery.etcd.io/new
-      discovery: https://discovery.etcd.io/<token>
-      # multi-region and multi-cloud deployments need to use $public_ipv4
-      addr: $public_ipv4:4001
-      peer-addr: $private_ipv4:7001
+  etcd2:
+    # generate a new token for each unique cluster from https://discovery.etcd.io/new
+    discovery: https://discovery.etcd.io/<token>
+    # multi-region and multi-cloud deployments need to use $public_ipv4
+    advertise-client-urls: http://$public_ipv4:2379
+    initial-advertise-peer-urls: http://$private_ipv4:2380
+    listen-client-urls: http://0.0.0.0:2379
+    listen-peer-urls: http://$private_ipv4:2380
```

Comment:
I vote yes but with a comment saying the old ports are only there for
compatibility with older configurations and software that still defaults to
the old client port. The server ports are less important but also worth
calling out in case someone is deploying new coreos systems with existing
firewall rules.
On Apr 6, 2015 10:58 AM, "Brandon Philips" <notifications@github.com> wrote:

> In Documentation/cloud-config.md
> <https://github.com/coreos/coreos-cloudinit/pull/330#discussion_r27815083>
> :
>
> >  If the platform environment supports the templating feature of coreos-cloudinit it is possible to automate etcd configuration with the `$private_ipv4` and `$public_ipv4` fields. For example, the following cloud-config document...
> >
> >  ```yaml
> >  #cloud-config
> >
> >  coreos:
> > -  etcd:
> > -      name: node001
> > -      # generate a new token for each unique cluster from https://discovery.etcd.io/new
> > -      discovery: https://discovery.etcd.io/<token>
> > -      # multi-region and multi-cloud deployments need to use $public_ipv4
> > -      addr: $public_ipv4:4001
> > -      peer-addr: $private_ipv4:7001
> > +  etcd2:
> > +    # generate a new token for each unique cluster from https://discovery.etcd.io/new
> > +    discovery: https://discovery.etcd.io/<token>
> > +    # multi-region and multi-cloud deployments need to use $public_ipv4
> > +    advertise-client-urls: http://$public_ipv4:2379
> > +    initial-advertise-peer-urls: http://$private_ipv4:2380
> > +    listen-client-urls: http://0.0.0.0:2379
> > +    listen-peer-urls: http://$private_ipv4:2380
>
> we should tell people to add both 2380/7001 and 2379/4001, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/coreos-cloudinit/pull/330/files#r27815083>.
>